tasks/monitoring.yml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Fail if package state is not supported
+  fail:
+    msg: "The Cloud Ops Ansible role only supports the following package states: \"present\", and \"absent\"."
+  when: package_state not in ["present", "absent"]
+
+- block:
+  - name: Download script
+    get_url:
+      url: https://dl.google.com/cloudagents/add-monitoring-agent-repo.sh
+      dest: /tmp/add-monitoring-agent-repo.sh
+      mode: 0755
+
+  - name: Add repo
+    command:
+      chdir: /tmp
+      cmd: bash add-monitoring-agent-repo.sh
+    notify: Restart Monitoring Agent
+
+  when: package_state == "present"
```

Comment:
This means you don't try to clean up the repo when package_state is set to absent... maybe leave that as a TODO comment if it's nontrivial?

Reply:
Unfortunately theres no generic repository manager module, so this will require distro specific logic (which we were trying to avoid by using the shell script).

Reply:
Looking into this further, this requires deriving the repo name. If so, I don't think invoking the shell script is actually saving us in terms of complexity/maintenence. Unless of course we modify the script to support uninstalls.